src/template_parser.rs

```diff
@@ -906,6 +906,20 @@ fn build_string_method<'a, L: TemplateLanguage<'a>>(
                 TemplatePropertyFn(|s: &String| s.lines().next().unwrap_or_default().to_string()),
             ))
         }
+        "upper" => {
```

Comment:
nit: at some point we should start using `const &'static str`, we have a risk of forgetting to update a string literal potentially used at several places.

Reply:
I think it's OK to mostly relate on tests for this. Moreover, I don't think this particular string is repeated in the code (as opposed to tests or docs) in a way where it has to match exactly.